tests/functional/workloads/cnv/test_vm_hotpl_snap_clone.py

```diff
@@ -0,0 +1,238 @@
+            new_rbd_pool=True,
+            mapOptions="krbd:rxbounce",
+            mounter="rbd",
+        )
+
+        # Create ceph-csi-kms-token in the tenant namespace
+        proj_obj = project_factory()
+        kms.vault_path_token = kms.generate_vault_token()
+        kms.create_vault_csi_kms_token(namespace=proj_obj.namespace)
+        pvk_obj = PVKeyrotation(sc_obj_def)
+        pvk_obj.annotate_storageclass_key_rotation(schedule="*/3 * * * *")
+
+        file_paths = ["/source_file.txt", "/new_file.txt"]
+
+        # Create a PVC-based VM (VM1)
+        vm_obj_pvc = cnv_workload(
+            storageclass=sc_obj_def.name,
+            namespace=proj_obj.namespace,
+            volume_interface=constants.VM_VOLUME_PVC,
+        )
+
+        # Create the PVC for VM1
+        pvc_obj = pvc_factory(
+            project=proj_obj,
+            storageclass=sc_obj_def,
+            size=20,
+            access_mode=constants.ACCESS_MODE_RWX,
+            volume_mode=constants.VOLUME_MODE_BLOCK,
+        )
+        log.info(f"PVC {pvc_obj.name} created successfully")
+
+        # Create a DVT-based VM (VM2)
+        vm_obj_dvt = cnv_workload(
+            storageclass=sc_obj_def.name,
+            namespace=proj_obj.namespace,
+            volume_interface=constants.VM_VOLUME_DVT,
+        )
+
+        # Create the PVC for VM2
+        dvt_obj = pvc_factory(
+            project=proj_obj,
+            storageclass=sc_obj_def,
+            size=20,
+            access_mode=constants.ACCESS_MODE_RWX,
+            volume_mode=constants.VOLUME_MODE_BLOCK,
+        )
+        log.info(f"PVC {dvt_obj.name} created successfully")
+
+        # List of VM-PVC pairs for hotplug testing
+        vms_pvc = [(vm_obj_pvc, pvc_obj), (vm_obj_dvt, dvt_obj)]
+        before_disks_hotplug = []
+
+        # Hotplug disks and perform I/O operations
+        for i, (vm_obj, pvc) in enumerate(vms_pvc):
+            # Verify disks before hotplugging
+            before_disks = vm_obj.run_ssh_cmd("lsblk -o NAME,SIZE,MOUNTPOINT -P")
+            log.info(f"Disks before hotplug on VM {vm_obj.name}:\n{before_disks}")
+            before_disks_hotplug.append(before_disks)
+
+            # Hotplug the PVC volume to the VM
+            log.info(f"Hotplugging PVC {pvc.name} to VM {vm_obj.name}")
+            vm_obj.addvolume(volume_name=pvc.name)
+
+            # Wait for the disk to be hotplugged successfully
+            sample = TimeoutSampler(
+                timeout=600,
+                sleep=5,
+                func=verify_hotplug,
+                vm_obj=vm_obj,
+                disks_before_hotplug=before_disks,
+            )
+            sample.wait_for_func_value(value=True)
+            log.info(f"Hotplugged PVC {pvc.name} to VM {vm_obj.name}")
+
+            # Run I/O operation
+            log.info(f"Running I/O operation on VM {vm_obj.name}")
+            source_csum = run_dd_io(vm_obj=vm_obj, file_path=file_paths[0], verify=True)
```

Comment:
You need to ensure that you write the I/O to the hotplugged disk. To do this, you may need to create a filesystem on the hotplugged disk, write I/O to it, and verify the data integrity after the clone/snapshot operation.